nsqd/protocol_v2.go

```diff
@@ -504,7 +504,7 @@ func (p *protocolV2) AUTH(client *clientV2, params [][]byte) ([]byte, error) {
 		return nil, protocol.NewFatalClientErr(err, "E_BAD_BODY", "AUTH failed to read body")
 	}
 
-	if client.HasAuthorizations() {
+	if client.ClientAuth.HasAuthorizations() {
```

Comment:
I agree it doesn't look pleasant. I was thinking of `ClientAuth` as a service which `clientV2` depends on. I'm not sure if that makes it a good candidate to embed as part of the type or if better naming would address the issue while making the separation clear.